pkg/providers/instance/instance.go

```diff
@@ -212,26 +214,50 @@ func (p *Provider) newNetworkInterfaceForVM(vmName string, backendPools *loadbal
 		enableAcceleratedNetworking = true
 	}
 
-	return armnetwork.Interface{
+	nic := armnetwork.Interface{
 		Location: to.Ptr(p.location),
 		Properties: &armnetwork.InterfacePropertiesFormat{
 			IPConfigurations: []*armnetwork.InterfaceIPConfiguration{
 				{
 					Name: &vmName,
 					Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
-						Primary:                   to.Ptr(true),
-						PrivateIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodDynamic),
+						Primary:                   lo.ToPtr(true),
+						PrivateIPAllocationMethod: lo.ToPtr(armnetwork.IPAllocationMethodDynamic),
 						Subnet: &armnetwork.Subnet{
 							ID: &p.subnetID,
 						},
 						LoadBalancerBackendAddressPools: ipv4BackendPools,
 					},
 				},
 			},
-			EnableAcceleratedNetworking: to.Ptr(enableAcceleratedNetworking),
-			EnableIPForwarding:          to.Ptr(true),
+			EnableAcceleratedNetworking: lo.ToPtr(enableAcceleratedNetworking),
+			EnableIPForwarding:          lo.ToPtr(true),
 		},
 	}
+	networkPlugin := options.FromContext(ctx).NetworkPlugin
+	networkPluginMode := options.FromContext(ctx).NetworkPluginMode
+	if networkPlugin == consts.NetworkPluginAzure && networkPluginMode != consts.PodNetworkTypeOverlay {
+		// NOTE: We don't need to set LoadBalancerBackendAddressPools for secondary ip configs.
+		// AzureCNI without overlay as pod networking type requires we configure additional secondary ips
+		// TODO: When MaxPods comes from the AKSNodeClass kubelet configuration, get the number of secondary
+		// ips from the nodeclass instead of using the default
+		for i := 1; i < utils.DefaultMaxPods(networkPlugin)-consts.StaticAzureCNIHostNetworkAddons; i++ {
```

Comment:
I am very tempted here to just go ahead and implement maxpods as a setting in the AKS Nodeclass under a new kubeletconfiguration but exercised restraint. 

This feels gross to me. Using defaults here is kind of icky even though its the pattern we have been following thus far.